ciao-controller/openstack_compute.go

```diff
@@ -667,5 +668,13 @@ func (c *controller) startComputeService() error {
 	// start service.
 	service := fmt.Sprintf(":%d", computeAPIPort)
 
-	return http.ListenAndServeTLS(service, httpsCAcert, httpsKey, r)
+	glog.Infof("Starting compute API on port %d", computeAPIPort)
+
+	server := &http.Server{
+		Handler: r,
+		Addr:    service,
+	}
+
+	c.httpServers = append(c.httpServers, server)
```

Comment:
I think this might be racy as all the methods that start a service are run in their own go routines.